pkg/kubelet/api/v1alpha1/runtime/api.pb.go

```diff
@@ -2697,26 +2697,36 @@ type RuntimeServiceClient interface {
 	// RunPodSandbox creates and starts a pod-level sandbox. Runtimes should ensure
 	// the sandbox is in ready state.
 	RunPodSandbox(ctx context.Context, in *RunPodSandboxRequest, opts ...grpc.CallOption) (*RunPodSandboxResponse, error)
-	// StopPodSandbox stops the running sandbox. If there are any running
-	// containers in the sandbox, they should be forcibly terminated.
+	// StopPodSandbox stops any running process that is part of the sandbox and
+	// reclaims network resources (e.g., IP addresses) allocated to the sandbox.
+	// It is recommended to stop all containers in the sandbox before stopping
+	// the sandbox itself. If there are any running containers in the sandbox,
+	// they should be forcibly terminated.
+	// This call is idempotent, and should return no error if all relevant
+	// resources have already been reclaimed. It is expected to be called
+	// multiple times before the sandbox removal.
```

Comment:
please clarify that the shim will receive at least one such call per sandbox reported to have exited by the runtime, and 0 calls if the sandbox is deleted in such a way that the runtime is unable to report it to the kubelet (or whatever the case may be).